src/components/WalletModal/Option.test.tsx

```diff
@@ -13,7 +13,7 @@ jest.mock('components/AccountDrawer')
 
 beforeEach(() => {
   jest.spyOn(console, 'debug').mockReturnValue()
-  mocked(useAccountDrawer).mockReturnValue([true, mockToggleDrawer])
+  mocked(useToggleAccountDrawer).mockReturnValue(mockToggleDrawer)
```

Comment:
I think you should add an add'l assertion in the test that this is actually called.
That it was missing is why you felt comfortable changing `onSuccess` to optional, before.